spec/system/v2_lesson_project_submissions/add_submission_spec.rb

```diff
@@ -55,8 +54,33 @@
     end
   end
 
+  context 'when lesson does not allow previews' do
+    let(:lesson) { create(:lesson, :project, has_live_preview: false) }
+    let(:user) { create(:user) }
+
+    before do
+      sign_in(user)
+      visit lesson_path(lesson)
+    end
+
+    it 'adds the submission without a preview' do
+      Pages::ProjectSubmissions::Form
+        .new(has_live_preview: false)
+        .open
+        .fill_in
+        .submit
+
+      within(:test_id, 'submissions-list') do
+        expect(page).to have_content(user.username)
+        expect(page).to have_link('View code')
+        expect(page).not_to have_link('Live preview')
```

Comment:
I haven't done thing kind of thing much before, but it intrigues me. Do you find it generally helpful? Is there a risk of drifting away from reality, since there's such a test specific interface? Or is it more about convenience in this case? 

Curious to hear your thoughts in general on the goals / benefits / downsides of an approach like this

Reply:
Sorry @ChargrilledChook, I'm probably being very dense lol. But is this about the page object or the line you commented on `expect(page).not_to have_link('Live preview')`?

Reply:
No worries and my bad, re-reading my comment it is a bit cryptic / stream of consciousness :sweat_smile: 

You're spot on though, I'm referring to the page object `Pages::ProjectSubmissions::Form`